py-polars/src/npy.rs

```diff
@@ -32,7 +32,7 @@ pub unsafe fn aligned_array<T: Element + NativeType>(
     let len = buf.len();
     let buffer_ptr = buf.as_mut_ptr();
 
-    let dims = [len].into_dimension();
+    let mut dims = [len].into_dimension();
```

Comment:
Do we need to motivate the `mut` qualifier here?

Reply:
Why do we need a mut here?

Reply:
This was suggested by the compiler.

```
error[E0596]: cannot borrow `dims` as mutable, as it is not declared as mutable
  --> src/npy.rs:43:9
   |
43 |         dims.as_dims_ptr(),
   |         ^^^^^^^^^^^^^^^^^^ cannot borrow as mutable
   |
help: consider changing this to be mutable
   |
35 |     let mut dims = [len].into_dimension();
   |         +++
```

I guess it better be `const`-qualified instead (I am trying to see how I can type it).

Reply:
Alright, I think the mut is fine. The signatures would be explanatory enough upon inspection.